### PR TITLE
Add API route for set-last-dtt-green.

### PR DESCRIPTION
**Pull request by ashercodeorg**
A `/set-last-dtt-green` [slash command integration](https://codedotorg.slack.com/services/B4MDV9R7C) is being created to hit this new API route from Slack. This seems useful for marking robo-DTTs and manual DTTs as green.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
There is a significant amount of DRYing up that can be done to the tests. This is intentionally not being undertaken in this PR, I intend to do so in a follow-up PR.

File: pegasus/test/test_dev_routes.rb [hunk 1]
```diff
@@ -167,4 +168,49 @@ def in_rack_env(env)
...
+
+      CDO.slack_start_build_token = FAKE_SLACK_SLASH_TOKEN
+
+      @default_param = {
+        token: FAKE_SLACK_SLASH_TOKEN,
+        user_name: 'turing'
+      }
+    end
+
+    def in_rack_env(env)
```
Comment (ashercodeorg) [inline pegasus/test/test_dev_routes.rb:185]: The `in_rack_env` and `make_test_pegasus` methods are being shamelessly copied from above. As noted in the PR conversation, I intend to follow up with a PR that DRYs up this whole test suite.

File: pegasus/routes/dev_routes.rb [hunk 2]
```diff
@@ -27,3 +30,12 @@
...
   )
 end
+
+post '/api/dev/set-last-dtt-green' do
+  forbidden! unless rack_env == :test
+  dont_cache
+  forbidden! unless params[:token] == CDO.slack_set_last_dtt_green_token
+
+  sha = GitHub.sha('test')
+  InfraTestTopic.set_green_commit(sha)
```
Comment (islemaster) [inline pegasus/routes/dev_routes.rb:40]: Should this also set `DTT: yes` in the developers topic?  Or is that already handled by the dotd script so we don't want to do it here?

Comment (ashercodeorg) [reply pegasus/routes/dev_routes.rb:40]: That is handled by the DOTD script, though so is updating the Slack#infra-test topic. My vision for this API route usage is solely as a slack slash command. From that vantage point,  don't think updating `DTT: yes` is appropriate.

**Comment by ashercodeorg** [comment]
Just now, I copied the token provided by Slack [here](https://codedotorg.slack.com/services/B4MDV9R7C) into the `globals.yml` file on `test`.

**Comment by ashercodeorg** [comment]
Hmm... The new tests should have failed (as they did in the DTT). Did they not run as part of CircleCI?

**Comment by islemaster** [comment]
No, they definitely ran, and passed:
```
api/dev/set-last-dtt-green
  test_0001_is forbidden on non-test environments                 PASS (0.01s)
  test_0002_succeeds on test environment                          PASS (0.00s)
```

**Pull request merged**